lib/sinon/util/fake_xdomain_request.js

```diff
@@ -7,8 +7,24 @@
 /**
  * Fake XDomainRequest object
  */
+
+/**
+ * Returns the global to prevent assigning values to 'this' when this is undefined.
+ * This can occur when files are interpreted by node in strict mode.
+ * @private
+ */
+function getGlobal() {
+    "use strict";
+
+    return typeof window !== "undefined" ? window : global;
```

Comment:
The tests don't fail currently. I just tried it using `self`, that works too. Shouldn't there be a test case for the web workers? Currently the tests pass without using `self`, so that means there is a missing test case for web workers, isn't there?